app/app.go

```diff
@@ -1395,12 +1405,39 @@ func initParamsKeeper(appCodec codec.BinaryCodec, legacyAmino *codec.LegacyAmino
 	return paramsKeeper
 }
 
+func (app *App) setupUpgradeStoreLoaders() {
+	upgradeInfo, err := app.UpgradeKeeper.ReadUpgradeInfoFromDisk()
+	if err != nil {
+		panic(fmt.Sprintf("failed to read upgrade info from disk %s", err))
+	}
+
+	if app.UpgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
+		return
+	}
+
+	for _, upgrade := range Upgrades {
+		if upgradeInfo.Name == upgrade.UpgradeName {
+			app.SetStoreLoader(upgradetypes.UpgradeStoreLoader(upgradeInfo.Height, &upgrade.StoreUpgrades))
+		}
+	}
+
+}
+
 func (app *App) setupUpgradeHandlers() {
-	app.UpgradeKeeper.SetUpgradeHandler(
-		v1_beta3.UpgradeName,
-		v1_beta3.CreateUpgradeHandler(
-			app.mm,
-			app.configurator,
-			app.AccountKeeper,
-			app.BankKeeper))
+	for _, upgrade := range Upgrades {
+		app.UpgradeKeeper.SetUpgradeHandler(
+			upgrade.UpgradeName,
+			upgrade.CreateUpgradeHandler(
+				app.mm,
+				app.configurator,
+				app.BaseApp,
+				&app.AppKeepers,
+			),
+		)
+	}
+}
+
+func (app *App) setupAppkeeper() {
+	app.AppKeepers.AccountKeeper = &app.AccountKeeper
+	app.AppKeepers.BankKeeper = &app.BankKeeper
```

Comment:
I am not sure why it is doing update for AccountKeeper and BankKeeper only here - what about the others?

Reply:
I see ParamsKeeper and etc on AppKeepers as well

Reply:
This is to make the upgrade process common.
It is also a preparation to eventually make keeper setup like osmosis.

Reply:
Yeah I saw that it's similar to Osmosis but Osmosis does not have two account keepers on App. One on app.AppKeepers.AccountKeeper the other on app.AccountKeeper - I think it would be fine to remove app.AccountKeeper.